src/lib/github/proxy.ts

```diff
@@ -0,0 +1,100 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { githubApiPath, githubApiUrl } from '@/lib/github/constants';
+import { requireAuth } from '@/lib/auth/requireAuth';
+import { getBearerAccessToken } from '@/lib/auth/getBearerAccessToken';
+import * as Response from '@/lib/response';
+import { fetchRequest } from '@/lib/request';
+
+export async function proxy<T>(
+  request: NextRequest
+): Promise<NextResponse<Result<T>>> {
+  try {
+    const isAuthenticated = await requireAuth();
+    if (!isAuthenticated) return Response.Unauthorized;
+
+    const bearerToken = await getBearerAccessToken();
+    if (!bearerToken) return Response.Unauthorized;
+
+    // Security check: Only allow POST for GraphQL requests
+    const method = request.method;
+    if (method === 'POST') {
+      const requestUrl = new URL(request.url);
+      const pathname = requestUrl.pathname;
+      const githubPath = pathname.replace(githubApiPath, '');
+
+      if (githubPath !== '/graphql') {
+        return Response.MethodNotAllowed;
+      }
+    }
+
+    const headers = new Headers({
+      Accept: request.headers.get('accept') || 'application/vnd.github.v3+json',
+      Authorization: bearerToken,
+      'User-Agent': `${process.env.APP_NAME}/${process.env.APP_VERSION} (+${process.env.NEXT_PUBLIC_BASE_URL})`,
+    });
+
+    const requestUrl = new URL(request.url);
+    const pathname = requestUrl.pathname;
+    const githubPath = pathname.replace(githubApiPath, '');
+
+    const githubUrl = `${githubApiUrl}${githubPath}`;
+
+    const searchParams = requestUrl.searchParams.toString();
```

Comment:
Here too just to avoid the repetition 
```suggestion
    const { searchParams } = requestUrl;
```